openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/weighting/OptimizedPriorityWeighting.java

```diff
@@ -34,8 +34,8 @@ public OptimizedPriorityWeighting(FlagEncoder encoder, PMap map) {
 	}
 
 	@Override
-	public double calcWeight(EdgeIteratorState edgeState, boolean reverse, int prevOrNextEdgeId) {
-		double weight = super.calcWeight(edgeState, reverse, prevOrNextEdgeId);
+	public double calcWeight(EdgeIteratorState edgeState, boolean reverse, int prevOrNextEdgeId, long edgeEnterTime) {
+		double weight = super.calcWeight(edgeState, reverse, prevOrNextEdgeId, edgeEnterTime);
```

Comment:
What's with all the other weightings in `routing/graphhopper/extensions/weighting`? Do they not need this adaptation?

Reply:
Good point! This depends on whether a given weighting calls `super.calcWeight` and/or whether it depends on the edge speed. I will go over individual weightings and make adjustments if necessary.

Reply:
I've only changed `AdditionWeighting` and `FastestSafeWeighting`. `AdditionWeighting` is used to collate the so-called soft weightings such as `AccelerationWeighting`, `AvoidHillsWeighting`, `GreenWeighting`, `QuietWeighting` and `SteepnessDifficultyWeighting`. These are not "true" standalone weightings but rather provide a modifying factor applied to the base weighting. As such they do not need any adaptations because for the time being they are not time dependent, nor they rely on the speed calculator pipeline. `FastestSafeWeighting` is not currently being used in our setup, yet it is the weighting to which `ORSWeightingFactory` will default if no `shortest`, `fastest` or `recommended` weighting is specified and the flag encoder supports `PriorityWeighting`. 

Furthermore, no changes are necessary to `PreferencePriorityWeighting` or `ConstantWeighting` as the former one doesn't seem to be used in any scenario, while the latter one is defined just for the sake  of unit tests.

Last but not least, I've left `DistanceWeighting` as is. It should be fine for now, nevertheless in long run we might want to look into replacing it by regular `shortest` weighting which it effectively seems to be. I will open a separate issue for it later.